code/plugins/ccavenue/ccavenue/tmpl/default.php

```diff
@@ -22,7 +22,8 @@
 				<input type="hidden" name="amount" value="<?php echo (float)$vars->amount; ?>" />
 				<input type="hidden" name="order_id" value="<?php echo $vars->order_id; ?>" />
 				<input type="hidden" name="currency" value="<?php echo $vars->currency_code; ?>" />
-				<input type="hidden" name="redirect_url" value="<?php echo $vars->notify_url; ?>" />
+				<input type="hidden" name="redirect_url" value="<?php echo (empty($vars->url) ? $vars->notify_url : $vars->url ); ?>" />
```

Comment:
There should be no logic in the layout. Move the condition to the plugin file.

Reply:
Yes, I have added a condition in the plugin for this and all HTML rendered from the plugin for payment process.